onnx/defs/tensor/utils.cc

```diff
@@ -96,20 +96,28 @@ void resizeShapeInferenceVersioned(InferenceContext& ctx, int opset_version) {
   bool hasSizesInput = ctx.hasInput(3);
 
   const TensorProto* scales = 2 < ctx.getNumInputs() ? ctx.getInputData(2) : nullptr;
-  const TensorProto* sizes = 3 < ctx.getNumInputs() ? ctx.getInputData(3) : nullptr;
+  std::vector<int64_t> sizes_data;
+  if (3 < ctx.getNumInputs()) {
+    bool found_sizes = false;
+    const auto sizes_shape = getShapeInput(ctx, 3, found_sizes);
+    // If sizes is an empty shape, assume it's not provided
```

Comment:
Can you clarify why? Specifically: is my understanding correct that this is an error-situation? Or, is this supposed to be a valid situation? (I guess it could be valid if `axes` is empty ... except for that edge-case.)

Reply:
Oh ... I remember, is this the old test-case fix where empty-tensor was used (before input was made optional)?

Reply:
https://github.com/onnx/onnx/pull/4388 I think it just provides some flexibility to let ONNX handle old models.

>  I remember, is this the old test-case fix where empty-tensor was used (before input was made optional)?

Yes! Some existing models in ONNX Model Zoo have this kind of usage.